src/RequestInformation.php

```diff
@@ -147,25 +146,75 @@ public function setStreamContent(StreamInterface $value): void {
     }
 
     /**
-     * Sets the request body from a model with the specified content type.
+     * Sets the request body from a model using the specified content type.
+     *
      * @param RequestAdapter $requestAdapter The adapter service to get the serialization writer from.
      * @param string $contentType the content type.
-     * @param Parsable ...$values the models.
+     * @param Parsable $value the models.
      */
-    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable ...$values): void {
-        if (empty($values)) {
-            throw new InvalidArgumentException('$values cannot be empty.');
+    public function setContentFromParsable(RequestAdapter $requestAdapter, string $contentType, Parsable $value): void {
+        try {
+            $writer = $requestAdapter->getSerializationWriterFactory()->getSerializationWriter($contentType);
+            $writer->writeObjectValue(null, $value);
+            $this->headers[self::$contentTypeHeader] = $contentType;
+            $this->content = $writer->getSerializedContent();
+        } catch (Exception $ex) {
```

Comment:
Could there be a way of making the variable name for the exception more meaningful in the context, and if not possible just something like exception//exception_message etc